packages/astro/src/runtime/server/serialize.ts

```diff
@@ -85,6 +88,15 @@ function convertToSerializedForm(
 		case '[object Array]': {
 			return [PROP_TYPE.JSON, JSON.stringify(serializeArray(value, metadata, parents))];
 		}
+		case '[object Uint8Array]': {
+			return [PROP_TYPE.Uint8Array, JSON.stringify(serializeArray(Array.from(value as Uint8Array), metadata, parents))];
```

Comment:
Do each of these need to use `serializeArray`? Every value will always be an int, so I think this should be fine. Note that the logic in `astro-island.ts` will need to be updated as well.

```suggestion
			return [PROP_TYPE.Uint8Array, JSON.stringify(Array.from(value as Uint8Array))];
```

Reply:
Skipping the `serializeArray` will effectively halve the payload size, so I think it's worth doing!

Reply:
Feel free to cherry-pick [46b6894](https://github.com/withastro/astro/commit/46b68943ede7667d134efeb0a5525557f9b84ef9) directly if you want!

Reply:
Your suggestion makes perfect sense, I cherry-picked it. Now it serializes/hydrates in a smarter way!